notification/src/main/java/com/iluwatar/RegisterWorker.java

```diff
@@ -0,0 +1,78 @@
+package com.iluwatar;
+
+import java.time.LocalDate;
+import java.time.Period;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Class which handles actual internal logic and validation for worker registration.
+ * Part of the domain layer which collects information and sends it back to the presentation.
+ */
+@Slf4j
+public class RegisterWorker extends ServerCommand {
+  protected RegisterWorker(RegisterWorkerDto worker) {
+    super(worker);
+  }
+
+  /**
+   * Validates the data provided and adds it to the database in the backend.
+   */
+  public void run() {
+
+    validate();
+    if (!super.getNotification().hasErrors()) {
+      LOGGER.info("Register worker in backend system");
+    }
+  }
+
+  /**
+   * Validates our data. Checks for any errors and if found, adds to notification.
+   */
+  private void validate() {
+    var ourData = ((RegisterWorkerDto) this.data);
+    //check if any of submitted data is not given
+    // passing for empty value validation
+    fail(isNullOrBlank(ourData.getName()), RegisterWorkerDto.MISSING_NAME);
```

Comment:
I think we should throw exception instead of `fail`

Reply:
I don't think having exception handling for this. wanted it to as comparable with [this](https://martinfowler.com/eaaDev/Notification.html).